src/Betfair/Adapter/ArrayRpcAdapter.php

```diff
@@ -14,6 +14,21 @@ class ArrayRpcAdapter implements AdapterInterface
     public function adaptResponse($response)
     {
         $data = json_decode($response, true);
-        return $data['result'];
+        
+        // check if $data contains result from Betfair API
+        
+        if(isset($data['result'])){
+            
+        	return $data['result'];
+        	
+        }
+        
+        // check if $data contains error from Betfair API
+        
+        if(isset($data['error'])){
```

Comment:
Exactly, I have to find an elegant way to expose the credential (and add the ability to manipulate the token). It should be easy